sql/driver/conn.go

```diff
@@ -186,6 +188,10 @@ func (c *conn) Send(call sqlwire.Call) (*rows, error) {
 			} else if datum.StringVal != nil {
 				t[j] = *datum.StringVal
 			}
+			if t[j] != nil && !driver.IsValue(t[j]) {
+				err := fmt.Errorf("Value of bad type %v returned by database", reflect.TypeOf(t[j]))
```

Comment:
I think you can actually use a special verb `%T` to print the type of a thing. https://golang.org/pkg/fmt/